atecc608a_se.c

```diff
@@ -208,20 +245,64 @@ static psa_status_t atecc608a_asymmetric_sign(psa_key_slot_number_t key_slot,
     return status;
 }
 
+psa_status_t atecc608a_asymmetric_verify(psa_key_slot_number_t key_slot,
+                                         psa_algorithm_t alg,
+                                         const uint8_t *p_hash,
+                                         size_t hash_length,
+                                         const uint8_t *p_signature,
+                                         size_t signature_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    bool is_verified = false;
+
+    /* Keys 8 to 15 can store public keys. */
+    if (key_id < 7 || key_id > 15)
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    /* We can only do ECDSA on SHA-256 */
+    if (alg != PSA_ALG_ECDSA(PSA_ALG_SHA_256) && alg != PSA_ALG_ECDSA_ANY)
+    {
+        return PSA_ERROR_NOT_SUPPORTED;
+    }
+
+    if (hash_length != 32)
```

Comment:
Should we use the hash length macro here? `PSA_HASH_SIZE(PSA_ALG_SHA_256)`

May want to update the signature function as well at the same time, in a separate commit.

Reply:
Putting `PSA_HASH_SIZE` here may suggest that it has something to do with PSA. I would use a `ATCA_`-something macro, but I'm not sure if there's any... maybe `ATCA_SHA_DIGEST_SIZE`?

Reply:
I rethought it overnight, and I guess that `PSA_HASH_SIZE(PSA_ALG_SHA_256)` is better than a raw 32 - people who are more proficient than me will catch it immediately.